samples/javascript_nodejs/50.teams-messaging-extensions-search/bots/teamsMessagingExtensionsSearchBot.js

```diff
@@ -3,30 +3,113 @@
-            preview.content.tap = { type: 'invoke', value: { description: obj.package.description } };
-            const attachment = { ...heroCard, preview };
-            attachments.push(attachment);
-        });
+            case 'result grid': 
+                return this.GetResultGrid();
 
+            default: 
+                const response = await axios.get(`http://registry.npmjs.com/-/v1/search?${ querystring.stringify({ text: searchQuery, size: 8 }) }`);
+                
+                response.data.objects.forEach(obj => {
+                const heroCard = CardFactory.heroCard(obj.package.name);
+                const preview = CardFactory.heroCard(obj.package.name);
+                preview.content.tap = { type: 'invoke', value: { description: obj.package.description } };
+                const attachment = { ...heroCard, preview };
+                attachments.push(attachment);
+                });
+    
+                return {
+                    composeExtension: {
+                    type: 'result',
+                    attachmentLayout: 'list',
+                    attachments: attachments
+                    }
+                };
+            }       
+        }
+
+     GetAdaptiveCard() {
+
+        const preview = CardFactory.thumbnailCard(
+            'Adaptive Card',
+            'Please select to get the card'
+        );
+
+        const adaptive = CardFactory.adaptiveCard(AdaptiveCard);
+        
+        const attachment = { ...adaptive,preview };
         return {
             composeExtension: {
-                type: 'result',
-                attachmentLayout: 'list',
-                attachments: attachments
+            type: 'result',
+            attachmentLayout: 'list',
+            attachments: [attachment]
             }
         };
     }
+    
+    GetConnectorCard() {    
+        const preview = CardFactory.thumbnailCard(
+        'Connector Card',
+        'Please select to get the card'
+        );
+
+        const connector = CardFactory.o365ConnectorCard(ConnectorCard);
+        const attachment = {...connector,preview}
+        return {
+            composeExtension: {
+            type: 'result',
+            attachmentLayout: 'list',
+            attachments: [attachment]
+            }
+        };
+    }
+   
+    GetResultGrid() {
+        const attachments = [];    
+        var files = fs.readdirSync(publicDir,function(err, result) {
+            if(err) {    
+               console.log('error', err);
+                }
+            });
+
+        var grid="";
+        files.forEach(file=>{
+        grid = CardFactory.thumbnailCard(
+            '',
+            [{ url: `${baseurl}/Images/${file}` }]
+        );
+        attachments.push(grid);
+       });
+       
+        return {
+            composeExtension: {
+            type: 'result',
+            attachmentLayout: 'grid',
+            attachments: attachments
+                }
+            };     
+        }
```

Comment:
I think the spacing can be cleaned up here, as well. Tough to say for sure looking at this in GitHub.